src/router/mod.rs

```diff
@@ -0,0 +1,128 @@
+mod tree;
+
+use Context;
+use Handler;
+use Request;
+use Result;
+use Route;
+use Tween;
+
+use std::rc::Rc;
+
+use hyper::Response;
+
+#[derive(Clone)]
+pub struct Router<S: Clone> {
+    handlers: tree::Tree<S>,
+    tweens: Vec<Tween<S>>,
+}
+
+impl<S: Clone + 'static> Router<S> {
+    pub fn new() -> Router<S> {
+        Router {
+            handlers: tree::Tree::new(),
+            tweens: vec![],
+        }
+    }
+
+    pub fn get(mut self, pattern: &str, handler: Handler<S>) -> Router<S> {
+        let trimmed = trim_path(pattern);
+        let segments = trimmed.split("/");
+        let mut current = 0;
+
+        for segment in segments {
+            if segment.starts_with(":") || segment.starts_with("*") {
+                current = self.handlers.node_set_wildcard(
+                    current,
+                    String::from(segment)
+                );
+            } else {
+                current = self.handlers.node_add_child(
+                    current,
+                    String::from(segment)
+                );
+            }
+        }
+
+        self.handlers.node_set_handler(current, handler);
```

Comment:
At some point this should be an error if a handler is already set

Reply:
I fixed this now but it's in the `node_set_handler` method.